projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -55,96 +59,182 @@ export class GenericFilterComponent {
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.minSelectedError);
-          continue;
+  private buildForm(config: FilterConfig, settings: FilterSetting, reset?: boolean): void {
+    const findPersonControl = this.form ? this.form.get('findPersonControl') : null;
+    this.form = this.fb.group({});
+    if (findPersonControl) {
+      // in order to maintain find person component, control needs to be kept
+      this.form.addControl('findPersonControl', findPersonControl);
+    }
+    for (const field of config.fields) {
+      if (field.type === 'checkbox') {
+        const formArray = this.buildCheckBoxFormArray(field, settings);
+        this.form.addControl(field.name, formArray);
+      } else {
+        const validators: ValidatorFn[] = [];
+        if (field.minSelected && field.minSelected > 0) {
+          validators.push(Validators.required);
         }
-
-        if (field.value.length > fieldConstraint.maxSelected) {
-          // display error for maximum selected
-          fieldConstraint.displayMaxSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.maxSelectedError);
-          continue;
+        let defaultValue: string = '';
+        if (reset && config.cancelSetting) {
+          const cancelField = config.cancelSetting.fields.find((f) => f.name === field.name);
+          defaultValue = cancelField ? cancelField.value[0] : '';
+        } else if (settings && settings.fields) {
+          const lastSavedValue = settings.fields.find((f) => f.name === field.name);
+          defaultValue = lastSavedValue ? lastSavedValue.value[0] as unknown as string : '';
         }
+        const control = new FormControl(defaultValue, validators);
+        this.form.addControl(field.name, control);
+      }
+    }
+  }
 
-        if (field.value.length < fieldConstraint.minSelected) {
-          // display error for minimum selected
-          fieldConstraint.displayMinSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.minSelectedError);
-          continue;
-        }
-        fieldConstraint.displayMinSelectedError = false;
-        fieldConstraint.displayMaxSelectedError = false;
+  private buildCheckBoxFormArray(field: FilterFieldConfig, settings: FilterSetting): FormArray {
+    const formArray = this.fb.array([], [
+      minSelectedValidator(field.minSelected),
+      maxSelectedValidator(field.maxSelected)
+    ]);
+    let defaultValues;
+    if (settings && settings.fields) {
+      defaultValues = settings.fields.find((f) => f.name === field.name);
+    }
+    for (const option of field.options) {
+      let checked = false;
+      if (defaultValues && Array.isArray(defaultValues.value)) {
+        checked = !!defaultValues.value.find((value) => value === option.key);
+      }
+      formArray.push(new FormControl(checked));
+    }
+    return formArray;
+  }
```

Comment:
Wouldn't this break if you have multiple checkbox fields? Indices would start from 0 again, so you end up with duplicate formfield names

Reply:
I don't think so, the form will be build in this  ` { locations: [false, false, false], roles: [false, false, false]`; So as long as you have different names for the control names it doesn't matter.